File: python/refactoring/grading.py
def get_letter_grade(grade: int) -> str:
    match grade:
        case 1:
            return "A"
        case 2:
            return "B"
        case 3:
            return "C"
        case 4:
            return "D"
        case 5:
            return "E"
        case 6:
            return "F"
        case _:
            raise ValueError(f"unknown grade {grade}")

File: csharp/Refactoring/Grading.cs
namespace Refactoring;

public class Grading
{
    public static char GetLetterGrade(int grade)
    {
        if(grade >= 1 && grade <= 6)
        {
            return (char)(64 + grade);
        } else
        {
            throw new ArgumentException("unknown grade " + grade);
        }
    }
}
